### PR TITLE
Add new preboot configuration vars

### PR DESCRIPTION
**Pull request by ghost**
New vars are:
- configuration_class_name
- configuration_module
- configuration_path
- service_env 

Update tooltip info next to `configuration` field on `preboot coniguration` details page.
Add a few paragraphs of deployment doc. and tests.

### CODE REVIEW

File: docs/user/deployment.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    - Type (one of these options: 'kickstart`, `iPXE`)
+    - Configuration
+    - Description
+
+
+Ad. `Configuration` field:
+This field allows you to write `kickstart` (or `iPXE`) configuration.
+It's possible to include variables from Ralph. These are:
+
+    - configuration_path
```
Comment (mkurek) [inline docs/user/deployment.md]: could you add some example of each variable?

File: src/ralph/deployment/views.py [hunk 2]
```diff
@@ -62,6 +63,7 @@ def _render_configuration(configuration, deployment):
             if deployment.obj.network_environment else ''
         ),
         'hostname': deployment.obj.hostname,
+        'service_env': str(deployment.obj.service_env),
```
Comment (mkurek) [inline src/ralph/deployment/views.py:85]: could we add service uid (deployment.obj.service.uid), configuration module (deployment.obj.configuration_path.module) and configuration class (deployment.obj.configuration_path.class_name) as well?

File: docs/user/deployment.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+Ad. `Configuration` field:
+This field allows you to write `kickstart` (or `iPXE`) configuration.
+It's possible to include variables from Ralph. These are:
+
+    - configuration_class_name (eg. 'www')
+    - configuration_module (eg. 'ralph')
+    - configuration_path (eg. 'ralph/www')
+    - dc (eg. 'data-center1')
+    - deployment_id (eg. 'ea9ea3a0-1c4d-42b7-a19b-922000abe9f7')
+    - domain (eg. 'dc1')
```
Comment (mkurek) [inline docs/user/deployment.md]: dc1 -> dc1.mydc.net

File: docs/user/deployment.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    - done_url (eg. 'http://127.0.0.1:8000/deployment/ea9ea3a0-1c4d-42b7-a19b-922000abe9f7/mark_as_done')
+    - hostname (eg. 'ralph0.allegro.pl')
```
Comment (mkurek) [inline docs/user/deployment.md]: ralph0.allegro.pl -> ralph123.dc1.mydc.net

File: docs/user/deployment.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    - initrd (eg. 'http://127.0.0.1:8000/deployment/ea9ea3a0-1c4d-42b7-a19b-922000abe9f7/initrd')
+    - kernel (eg. 'http://127.0.0.1:8000/deployment/ea9ea3a0-1c4d-42b7-a19b-922000abe9f7/kernel')
+    - kickstart (eg. 'http://127.0.0.1:8000/deployment/ea9ea3a0-1c4d-42b7-a19b-922000abe9f7/kickstart')
+    - ralph_instance (eg. 'http://127.0.0.1:8000')
```
Comment (mkurek) [inline docs/user/deployment.md:39]: add some comment that this variable (and all hyperlinks here in general) are starting with ralph url defined in settings (which should not (or maybe should?) contain trailing slash)

File: docs/user/deployment.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    - service_env (eg. 'Backup systems - prod')
+    - service_uid (eg. 'sc-0')
```
Comment (mkurek) [inline docs/user/deployment.md]: sc-123

**Comment by ghost** [comment]
@mkurek trainling slash should be fixed, because now it's absent in rendered templates

**Comment by mkurek** [comment]
:ok:

File: src/ralph/deployment/models.py [hunk 7]
```diff
@@ -56,6 +56,27 @@ def autocomplete_str(self):
...
+<br>    - domain
+<br>    - done_url
+<br>    - hostname
+<br>    - initrd
+<br>    - kernel
+<br>    - kickstart
+<br>    - ralph_instance
+<br>    - service_env
+<br>    - service_uid
+""".strip()
```
Comment (ar4s) [inline src/ralph/deployment/models.py]: maybe extend variables by example of rendered values like you did in documentation?

**Comment by ar4s** [comment]
:ok:

**Pull request merged**